app/packages/core/src/components/Modal/use-looker.ts

```diff
@@ -0,0 +1,103 @@
+import * as fos from "@fiftyone/state";
+import React, { useEffect, useRef, useState } from "react";
+import { useErrorHandler } from "react-error-boundary";
+import { useRecoilValue } from "recoil";
+import { v4 as uuid } from "uuid";
+import { useClearSelectedLabels, useShowOverlays } from "./ModalLooker";
+import { useLookerOptionsUpdate } from "./hooks";
+import useKeyEvents from "./use-key-events";
+import { shortcutToHelpItems } from "./utils";
+
+const CLOSE = "close";
+
+function useLooker<L extends fos.Lookers>({
+  sample,
+}: {
+  sample: fos.ModalSample;
+}) {
+  const [id] = useState(() => uuid());
+  const initialRef = useRef<boolean>(true);
+  const ref = useRef<HTMLDivElement>(null);
+  const [reset, setReset] = useState(false);
+  const lookerOptions = fos.useLookerOptions(true);
+  const createLooker = fos.useCreateLooker(
+    true,
+    false,
+    lookerOptions,
+    undefined,
+    true
+  );
+  const selectedMediaField = useRecoilValue(fos.selectedMediaField(true));
+  const colorScheme = useRecoilValue(fos.colorScheme);
+  const looker = React.useMemo(() => {
+    reset;
+    selectedMediaField;
+    return createLooker.current(sample);
+  }, [createLooker, reset, sample, selectedMediaField]) as L;
+  const handleError = useErrorHandler();
+  const updateLookerOptions = useLookerOptionsUpdate();
+
+  fos.useEventHandler(looker, "clear", useClearSelectedLabels());
+  fos.useEventHandler(looker, "error", (event) => handleError(event.detail));
+  fos.useEventHandler(looker, "options", (e) => updateLookerOptions(e.detail));
+  fos.useEventHandler(looker, "reset", () => setReset((c) => !c));
+  fos.useEventHandler(looker, "select", fos.useOnSelectLabel());
+  fos.useEventHandler(looker, "showOverlays", useShowOverlays());
+
+  useEffect(() => {
+    !initialRef.current && looker.updateOptions(lookerOptions);
+  }, [looker, lookerOptions]);
+
+  useEffect(() => {
+    colorScheme;
```

Comment:
Same comment as above. A comment along the lines of "we want looker to update sample when color scheme updates" would make it a little more explicit for new devs.